app/src/displays/datetime/datetime.vue

```diff
@@ -2,88 +2,93 @@
-				return parse(props.value, 'HH:mm:ss', new Date());
+interface Props {
+	value: string;
+	type: 'dateTime' | 'date' | 'time' | 'timestamp';
+	format?: string;
+	relative?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+	format: 'long',
+	relative: true,
+});
+
+const { t } = useI18n();
+
+const displayValue = ref<string | null>(null);
+
+const localValue = computed(() => {
+	if (!props.value) return null;
+
+	if (props.type === 'timestamp') {
+		return parseISO(props.value);
+	} else if (props.type === 'dateTime') {
+		return parse(props.value, "yyyy-MM-dd'T'HH:mm:ss", new Date());
+	} else if (props.type === 'date') {
+		return parse(props.value, 'yyyy-MM-dd', new Date());
+	} else if (props.type === 'time') {
+		return parse(props.value, 'HH:mm:ss', new Date());
+	}
+
+	return null;
+});
+
+const relativeFormat = (value: Date) =>
+	localizedFormatDistance(value, new Date(), {
+		addSuffix: true,
+	});
+
+watch(
+	localValue,
+	async (newValue) => {
+		if (newValue === null) {
+			displayValue.value = null;
+			return;
+		}
+
+		if (props.relative) {
+			displayValue.value = await relativeFormat(newValue);
+		} else {
+			let format;
+			if (props.format === 'long') {
+				format = `${t('date-fns_date')} ${t('date-fns_time')}`;
+				if (props.type === 'date') format = String(t('date-fns_date'));
+				if (props.type === 'time') format = String(t('date-fns_time'));
+			} else if (props.format === 'short') {
+				format = `${t('date-fns_date_short')} ${t('date-fns_time_short')}`;
+				if (props.type === 'date') format = String(t('date-fns_date_short'));
+				if (props.type === 'time') format = String(t('date-fns_time_short'));
+			} else {
+				format = props.format;
 			}
 
-			return null;
-		});
-
-		const displayValue = ref<string | null>(null);
-
-		watch(
-			localValue,
-			async (newValue) => {
-				if (newValue === null) {
-					displayValue.value = null;
-					return;
-				}
-
-				if (props.relative) {
-					displayValue.value = await localizedFormatDistance(newValue, new Date(), {
-						addSuffix: true,
-					});
-				} else {
-					let format;
-					if (props.format === 'long') {
-						format = `${t('date-fns_date')} ${t('date-fns_time')}`;
-						if (props.type === 'date') format = String(t('date-fns_date'));
-						if (props.type === 'time') format = String(t('date-fns_time'));
-					} else if (props.format === 'short') {
-						format = `${t('date-fns_date_short')} ${t('date-fns_time_short')}`;
-						if (props.type === 'date') format = String(t('date-fns_date_short'));
-						if (props.type === 'time') format = String(t('date-fns_time_short'));
-					} else {
-						format = props.format;
-					}
-
-					displayValue.value = await localizedFormat(newValue, format);
-				}
-			},
-			{ immediate: true }
-		);
-
-		return { displayValue };
+			displayValue.value = await localizedFormat(newValue, format);
+		}
 	},
+	{ immediate: true }
+);
+
+const refreshInterval = ref<number | null>(null);
```

Comment:
Not that it would make much of a difference performance-wise, but `refreshInterval` can just be a regular value instead of a reactive one.

Reply:
good point! Was actually just using it out of habit 😄

Reply:
I know how that goes 😁 You get used to writing `ref`s everywhere pretty quickly!